### PR TITLE
WA tiers paused

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: queue/queue.go [hunk 1]
```diff
@@ -123,6 +123,14 @@ var luaPop = redis.NewScript(2, `-- KEYS: [EpochMS QueueType]
 
 	-- if we didn't find one, try again from our bulk queue
 	if not result[1] or isFutureResult then
+		-- check if we are paused for bulk queue
+		local pausedBulkKey = "paused_bulk:" .. queueName
```
Comment (rowanseymour) [inline queue/queue.go:127]: maybe we call this `rate_limit_bulk` so we have a way to rate limit everything and a way to rate limit just bulk sends

File: handlers/whatsapp/whatsapp.go [hunk 2]
```diff
@@ -1022,6 +1034,15 @@ func buildWhatsAppHeaders(channel courier.Channel) http.Header {
 	return header
 }
 
+func hasTiersError(payload mtErrorPayload) bool {
+	for _, err := range payload.Errors {
+		if err.Code == 471 && err.Title == "Spam rate limit hit" {
```
Comment (rowanseymour) [inline handlers/whatsapp/whatsapp.go:1039]: is it important to check error title? Just wondering if that can change

File: queue/queue_test.go [hunk 3]
```diff
@@ -60,6 +60,19 @@ func TestLua(t *testing.T) {
 	delay := time.Second*2 - time.Duration(time.Now().UnixNano()%int64(time.Second))
 	time.Sleep(delay)
 
+	conn.Do("set", "paused_bulk:chan1", "engaged")
+	conn.Do("EXPIRE", "paused_bulk:chan1", 5)
```
Comment (rowanseymour) [inline queue/queue_test.go:64]: let's start a convention of using UPPERCASE for redis commands

**Review by rowanseymour — CHANGES_REQUESTED** [review]

File: queue/queue.go (hunk 1 above)
Comment (norkans7) [reply queue/queue.go:127]: so replace `paused_bulk` with `rate_limit_bulk`?

File: handlers/whatsapp/whatsapp.go (hunk 2 above)
Comment (norkans7) [reply handlers/whatsapp/whatsapp.go:1039]: I can remove that

File: queue/queue.go (hunk 1 above)
Comment (rowanseymour) [reply queue/queue.go:127]: yeah

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**